javaCreateAccount/src/main/java/com/hamdan/forzenbook/java/createaccount/JavaCreateAccountActivity.java

```diff
@@ -40,13 +39,13 @@ protected void onCreate(Bundle savedInstanceState) {
         FrameLayout view = binding.getRoot();
         setContentView(view);
 
-
         binding.createAccountSubmitButton.setEnabled(false);
         binding.createAccountClickBlocker.setOnClickListener(blank -> {
         });  //purposely empty to prevent clicks when its enabled
 
+        binding.createAccountLayoutToolBar.setNavigationOnClickListener(listener -> model.backButtonPressed(this));
 
-        binding.createAccountSubmitButton.setOnClickListener(listener -> model.createAccount());
+        binding.createAccountSubmitButton.setOnClickListener(listener -> model.createAccount(this));
```

Comment:
Please make this `createAccountClicked` so that the viewmodel keeps the input functions as ui events